cdatgui/cdat/plotter.py

```diff
@@ -156,24 +157,33 @@ def can_plot(self):
 
     @property
     def canvas(self):
-        # print "MANAGER CANVAS:", self.source.canvas
         return self.source.canvas
 
     def gm(self):
-        return self._gm
+        return vcs.getgraphicsmethod(self._type, self._gm)
 
     def set_gm(self, gm):
-        # check gm vs vars
-        self._gm = gm
-        if self.can_plot():
+        plot = True
+        if isinstance(gm, tuple):
+            plot = gm[1]
+            gm = gm[0]
+
+        self._gm = gm.name
+        self._type = vcs.graphicsmethodtype(gm)
+        if plot and self.can_plot():
             self.plot()
+        self.source.gm_label.setText(self._gm)
 
     graphics_method = property(gm, set_gm)
 
     def get_vars(self):
         return self._vars
 
     def set_vars(self, v):
+        plot = True
```

Comment:
Right, OK, I remember now. So, what I think we should do is just remove the automatic replotting when plotter attributes are set, and manually call plot from the places where we want it to plot (whenever we drop on a variable, GM, or template, and in the Tab.replotPlottersUpdateVars function, after it's finished updating everything).